source/Color.cpp

```diff
@@ -74,7 +80,58 @@ bool Color::IsLoaded() const
 // Get a float vector representing this color, for use by OpenGL.
 const float *Color::Get() const
```

Comment:
Doing this in `Get` seems wrong to me. Additionally, calling `new` without ever deleting the created object leaks memory, which in this case would be every frame.

I would suggest making a new function that returns a `Color` object that has the filter applied, and then in the shaders you would call that function to get the actual color to draw.